mlrun/runtimes/kubejob.py

```diff
@@ -240,17 +240,6 @@ def deploy(
                 state = self._build_watch(watch, show_on_failure=show_on_failure)
                 ready = state == "ready"
                 self.status.state = state
-        else:
```

Comment:
🔥 
I plan to move the deploy to the launchers and have it raise an exception if called from the server side FYI

Reply:
With _run it is more obvious why to the runtime handler, but for the deploy there is a debate, I suggest we discuss it offline